include/deal.II/trilinos/nox.templates.h

```diff
@@ -961,6 +961,7 @@ namespace TrilinosWrappers
     n_jacobian_applications  = 0;
     n_nonlinear_iterations   = 0;
     n_last_linear_iterations = 0;
+    pending_exception        = nullptr;
```

Comment:
Needed to be able to reuse the solver.

Reply:
This doesn't quite seem right. If there is still an exception pending, we should assert. Can you create a testcase that shows how this can happen?